doc/user-manual/language/syntactic-sugar.lagda.rst

```diff
@@ -61,6 +61,22 @@ Syntactic Sugar
    :depth: 2
    :local:
 
+.. _hidden_argument_puns:
+
+Hidden argument puns
+====================
+
+If the option :option:`--hidden-argument-puns` is used, then the
+pattern ``{x}`` is interpreted as ``{x = x}``, and the pattern
+``⦃ x ⦄`` is interpreted as ``⦃ x = x ⦄``. Here ``x`` must be an
+unqualified name that does not refer to a constructor that is in
+scope: if ``x`` is qualified, then the pattern is not interpreted as a
+pun, and if ``x`` is unqualified and refers to a constructor that is
+in scope, then the code is rejected.
+
+Note that ``{(x)}`` and ``⦃ (x) ⦄`` are not interpreted as puns, and
+that puns cannot be used in regular lambda abstractions.
+
```

Comment:
Make an extra paragraph for the lambda so it stands out, I suppose it should also be mentioned that puns are not active in Pi-types.